pkg/apis/scheduling/validation/validation.go

```diff
@@ -52,5 +61,10 @@ func ValidatePriorityClassUpdate(pc, oldPc *scheduling.PriorityClass) field.Erro
 	if pc.Value != oldPc.Value {
 		allErrs = append(allErrs, field.Forbidden(field.NewPath("Value"), "may not be changed in an update."))
 	}
+	// PreemptionPolicy is immutable and is checked by the ObjectMeta validator.
+	if utilfeature.DefaultFeatureGate.Enabled(features.NonPreemptingPriority) && *pc.PreemptionPolicy != *oldPc.PreemptionPolicy {
```

Comment:
This will panic if the field is nil. Use apivalidation.ValidateImmutableField instead. Also drop the feature gate check, validation here shouldn't care if the feature is enabled or not